pkg/didmethod/peer/generate.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	"crypto"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+const (
+	//numAlgo is the algorithm for choosing a numeric basis Reference: https://openssi.github.io/peer-did-method-spec/index.html#method-specific-identifier
+	numAlgo = "1"
+	//encAlgo  is the algorithm for encoding
+	encAlgo = "1"
+)
+
+//NewDid creates the peer DID. For example : did:peer:11-479cbc07c3f991725836a3aa2a581ca2029198aa420b9d99bc0e131d9f3e2cbe
+func NewDid(doc *did.Doc) (string, error) {
+	var peerDID string
+	peerPrefix := "did:peer:"
+	// doc MUST NOT include the DID
+	doc.ID = ""
+
+	if doc.PublicKey == nil || doc.Authentication == nil {
+		return "", fmt.Errorf("the genesis version must include public keys and authentication")
```

Comment:
should now use the xerrors package. see prior PR for an example.

Reply:
@talwinder50 here is the reference - https://github.com/hyperledger/aries-framework-go/pull/55/files#diff-d9e817deb772fd4e55da4b57afbca798R16

Reply:
done